src/org/mangui/hls/stream/StreamBuffer.as

```diff
@@ -1,1360 +1,1378 @@
-                    if ( (tagPos - firstPos) > maxDuration && ((isMain && tagIdx != mainIdx) || (!isMain && tagIdx != altIdx))) {
-                        // CONFIG::LOGGING {
-                        //     Log.info('discard level/sn/type/dts/pts:' + tag.fragLevel + '/' + tag.fragSN + '/' + tag.tag.typeString + '/' + tag.tag.dts + '/' + tag.tag.pts);
-                        // }
-                        break;
-                    }
-                }
-                // tag matches, shift it and push it in new array
-                tags.push(getNextTag(true));
-            }
-            return tags;
-        }
-
-        private function get min_pos() : Number {
-            if (audioExpected) {
-                if (videoExpected) {
-                    return Math.max(min_audio_pos, min_video_pos);
-                } else {
-                    return min_audio_pos;
-                }
-            } else {
-                return min_video_pos;
-            }
-        }
-
-        private function get min_min_pos() : Number {
-            if (audioExpected) {
-                if (videoExpected) {
-                    return Math.min(min_audio_pos, min_video_pos);
-                } else {
-                    return min_audio_pos;
-                }
-            } else {
-                return min_video_pos;
-            }
-        }
-
-        private function get min_audio_pos() : Number {
-            var min_pos_ : Number = Number.POSITIVE_INFINITY;
-            if (_audioTags.length) min_pos_ = Math.min(min_pos_, _audioTags[0].positionAbsolute - _liveSlidingMain );
-            return min_pos_;
-        }
-
-        private function get min_video_pos() : Number {
-            var min_pos_ : Number = Number.POSITIVE_INFINITY;
-            if (_videoTags.length) min_pos_ = Math.min(min_pos_, _videoTags[0].positionAbsolute - _liveSlidingMain );
-            return min_pos_;
-        }
-
-        private function get max_pos() : Number {
-            if (audioExpected) {
-                if (videoExpected) {
-                    return Math.min(max_audio_pos, max_video_pos);
-                } else {
-                    return max_audio_pos;
-                }
-            } else {
-                return max_video_pos;
-            }
-        }
-
-        private function get max_audio_pos() : Number {
-            var max_pos_ : Number = Number.NEGATIVE_INFINITY;
-            if (_audioTags.length) max_pos_ = Math.max(max_pos_, _audioTags[_audioTags.length - 1].positionAbsolute - _liveSlidingMain );
-            return max_pos_;
-        }
-
-        private function get max_video_pos() : Number {
-            var max_pos_ : Number = Number.NEGATIVE_INFINITY;
-            if (_videoTags.length) max_pos_ = Math.max(max_pos_, _videoTags[_videoTags.length - 1].positionAbsolute - _liveSlidingMain );
-            return max_pos_;
-        }
-
-        private 
```

Comment:
I wasn't sure if this should stay in, but the below clause `(position == -2)` should be there I think, I saw a very rare stutter without this logic (removing this would not effect the `canAppendTags` logic below).